### PR TITLE
[RKOTLIN-1083] Remove deprecated user methods

### PR DESCRIPTION
**Pull request by clementetb**
Closes https://github.com/realm/realm-kotlin/issues/1751

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ This release will bump the Realm file format from version 23 to 24. Opening a fi
 ### Breaking changes
 * Removed property `RealmLog.level`. Log levels can be set with `RealmLog.setLevel`. (Issue [#1691](https://github.com/realm/realm-kotlin/issues/1691) [JIRA](https://jira.mongodb.org/browse/RKOTLIN-1038))
 * Removed `LogConfiguration`. Log levels and custom loggers can be set with `RealmLog`. (Issue [#1691](https://github.com/realm/realm-kotlin/issues/1691) [JIRA](https://jira.mongodb.org/browse/RKOTLIN-1038))
+* Removed deprecated methods `User.identity` and `User.provider`, user identities can be accessed with `User.identities`. (Issue [#1751](https://github.com/realm/realm-kotlin/issues/1751) [JIRA](https://jira.mongodb.org/browse/RKOTLIN-1083))
+* `App.allUsers` now returns a list of users known locally. (Issue [#1751](https://github.com/realm/realm-kotlin/issues/1751) [JIRA](https://jira.mongodb.org/browse/RKOTLIN-1083))
```
Comment (rorbech) [inline CHANGELOG.md:10]:
```suggestion
* `App.allUsers` does no longer return a map, but only a list of users known locally. (Issue [#1751](https://github.com/realm/realm-kotlin/issues/1751) [JIRA](https://jira.mongodb.org/browse/RKOTLIN-1083))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,8 @@ This release will bump the Realm file format from version 23 to 24. Opening a fi
 ### Breaking changes
 * Removed property `RealmLog.level`. Log levels can be set with `RealmLog.setLevel`. (Issue [#1691](https://github.com/realm/realm-kotlin/issues/1691) [JIRA](https://jira.mongodb.org/browse/RKOTLIN-1038))
 * Removed `LogConfiguration`. Log levels and custom loggers can be set with `RealmLog`. (Issue [#1691](https://github.com/realm/realm-kotlin/issues/1691) [JIRA](https://jira.mongodb.org/browse/RKOTLIN-1038))
+* Removed deprecated methods `User.identity` and `User.provider`, user identities can be accessed with `User.identities`. (Issue [#1751](https://github.com/realm/realm-kotlin/issues/1751) [JIRA](https://jira.mongodb.org/browse/RKOTLIN-1083))
```
Comment (rorbech) [inline CHANGELOG.md:9]:
```suggestion
* Removed deprecated methods `User.identity` and `User.provider`, user identities can be accessed with the already existing `User.identities`. (Issue [#1751](https://github.com/realm/realm-kotlin/issues/1751) [JIRA](https://jira.mongodb.org/browse/RKOTLIN-1083))
```

**Review by rorbech — APPROVED** [review]
LGTM.

File: packages/test-sync/src/commonTest/kotlin/io/realm/kotlin/test/mongodb/common/AppTests.kt [hunk 3]
```diff
@@ -164,50 +163,51 @@ class AppTests {
...
         assertEquals(2, allUsers.size)
-        assertTrue(allUsers.containsKey(user3.identity))
+        assertTrue(allUsers.contains(user3))
 
         // Logging out users that registered with email/password will just put them in LOGGED_OUT state
         user3.logOut()
         allUsers = app.allUsers()
         assertEquals(2, allUsers.size)
-        assertTrue(allUsers.containsKey(user3.identity))
-        assertEquals(User.State.LOGGED_OUT, allUsers[user3.identity]!!.state)
```
Comment (nhachicha) [inline packages/test-sync/src/commonTest/kotlin/io/realm/kotlin/test/mongodb/common/AppTests.kt:189]: I think we should keep the LOGGED_OUT state check (maybe using `User.identities`?)

**Review by nhachicha — APPROVED** [review]

**Pull request merged**